de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/TreeViewController.java

```diff
@@ -0,0 +1,285 @@
+//					TreeViewUtilities.getInput("Enter attribute value"));
+		addAtrrOnIndex();
+		treeView.refresh();
+	}
+
+	void addAtrrOnIndex() {
+		int place = Integer.parseInt(TreeViewUtilities.getInput("Enter Place"));
+		AbstractAttribute attribute = (AbstractAttribute) treeView.getSelectionModel().getSelectedItem().getValue()
+				.getAttributes().get(place);
+		attribute.getAttributeValues().clear();
+		attribute.setAttributeKey(TreeViewUtilities.getInput("Enter attribute name"));
+		attribute.getAttributeValues().add(TreeViewUtilities.getInput("Enter attribute value"));
+
+	}
+
+	/**
+	 * 
+	 */
+	@FXML
+	void addChild() {
+		TreeItem<NodeUsage> newChild = new TreeItem<NodeUsage>();
+		newChild.setValue(new NodeUsage("DummyNode"));
+		contextMenu.hide();
+		try {
+			newChild.getValue().addAttribute("TEXT", TreeViewUtilities.getInput("Enter Child Text"));
+		} catch (NullPointerException e) {
+			return;
+		}
+		treeView.getSelectionModel().getSelectedItem().getChildren().add(newChild);
+		displayTotalNodeAmount();
+		treeView.refresh();
+	}
+
+	/**
+	 * 
+	 */
+	@FXML
+	void copy() {
+		contextMenu.hide();
+		copyList.clear();
+		copyList.addAll(treeView.getSelectionModel().getSelectedItems()); // Bug: Paste ist spiegelverkehrt bei oben
+																			// nach unten Markierung
+	}
+
+	@FXML
+	void paste() {
+
+		contextMenu.hide();
+
+		try {
+			for (TreeItem<NodeUsage> copiedNode : copyList) {
+				TreeItem<NodeUsage> t = new TreeItem<NodeUsage>();
+				t.setValue(copiedNode.getValue());
+				treeView.getSelectionModel().getSelectedItem().getParent().getChildren()
+						.add(treeView.getSelectionModel().getSelectedIndex(), t);
+				displayTotalNodeAmount();
+			}
+		} catch (NullPointerException e) {
+			return;
+		}
+	}
+
+	/**
+	 * 
+	 */
+	@FXML
+	void renameNode() { // Bug: �ndert nicht die Property sondern f�gt nur neue Property hinzu mit neuem
+						// Value
+		contextMenu.hide();
+		for (Attribute attribute : treeView.getSelectionModel().getSelectedItem().getValue().getAttributes()) {
+			if (attribute.getAttributeKey().toLowerCase().equals("name")) {
+				attribute.getAttributeValues().clear();
+			}
+		}
+		addAttribute("name", TreeViewUtilities.getInput("Enter new name"));
+		treeView.refresh();
+		services.eventBroker.send("nodePropertiesEvent", treeView.getSelectionModel().getSelectedItem().getValue());
+	}
+
+	@FXML
+	void cut() {
+		copy();
+		deleteNode();
+	}
+
+	@FXML
+	void extractToFile() {
+		List<TreeItem> extractList = new ArrayList<TreeItem>();
+		extractList.addAll(treeView.getSelectionModel().getSelectedItems());
+		TreeViewUtilities.extractTree(treeView, TreeViewUtilities.getInput("Extract to File"), extractList);
+	}
+
+	@FXML
+	void deleteNode() {
+		for (int i = 0; i < copyList.size(); i++) {
+			treeView.getSelectionModel().getSelectedItem().getParent().getChildren()
+					.remove(treeView.getSelectionModel().getSelectedItem());
+			displayTotalNodeAmount();
+		}
```

Comment:
deleteNode entfernt mehrere Nodes, umbenennung in deleteNodes?
i wird im For-Loop nicht vernwendet

Reply:
Methode wurde in der Zwischenzeit bereits überarbeitet